initial commit

rezvanf opened the pull request:
*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

mooretom commented:
Hi Ferdi,

Code looks good. I had only a couple of comments.

1 - Serverless/Serverless App with Dynamic Step Functions  /.aws-sam/build.toml
Line 5:
It’s a good idea not to have “/Users/rezvanf “ in the path as it leaks data. (Your user name) Also may cause users issues when they download the code and try to use it. Can that be a relative path?


2 - Serverless/Serverless App with Dynamic Step Functions  /functions/GetTargetImages/Function.cs
Line 67:
If there are no items will listObjectsResponse.S3Objects be null? Do we need to check for null there?

3 - Need to include a license file in the package. Usually we use the MIT license:

“
MIT License

Copyright (c) 2022 Thomas Moore

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
”


-

Thomas Moore
.NET Developer Advocate
Amazon Web Services<http://aws.amazon.com/>
E: ***@***.******@***.***>
M: +1 (508) 315-2567

[signature_315214956]<https://aws.amazon.com/>


Thoughts on our interaction? Provide Feedback here<https://feedback.aws.amazon.com/?ea=mooretom&fn=Tom&ln=Moore>.

From: Ferdi Rezvan ***@***.***>
Sent: Tuesday, August 2, 2022 6:59 AM
To: aws-samples/aws-net-guides ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [aws-samples/aws-net-guides] initial commit (PR #47)


Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/aws-samples/aws-net-guides/pull/47

Commit Summary

  *   37863fc<https://github.com/aws-samples/aws-net-guides/pull/47/commits/37863fcbb6ba96aaad9d8be1c7ac0cf860fe0fe6> initial commit

File Changes

(30 files<https://github.com/aws-samples/aws-net-guides/pull/47/files>)

  *   A Serverless/Serverless App with Dynamic Step Functions /.aws-sam/build.toml<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-9792bf0aa9a8456504af257b134776ade4ec7b98ff1a4f643ffe56f56d2415c7> (22)
  *   A Serverless/Serverless App with Dynamic Step Functions /README.md<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-cae2d591fd0f7fe2eed1e87a00d7bbfd649ab9c95206f383c12c11a7aa6e589e> (153)
  *   A Serverless/Serverless App with Dynamic Step Functions /dotnet_samples.svg<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-3a08a53fd9840edc1339bb42fe4297e27a393606e3dd9fe91f4e5117db15f144> (3)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/CompareImages/CompareImages.csproj<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-65095104a9d06213251d56e7d557f74138daf6ae34a1e366b0ef56f127c23274> (28)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/CompareImages/FaceComparison.cs<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-f036ee034488c34340a861b433c37b1e452b285b07153b0336d3b4788771ff59> (49)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/CompareImages/Function.cs<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-5de084b29cb9590844a10ec92700392c334bceeeee6784e29b07b78e832126fd> (37)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/CompareImages/ImageCompare.cs<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-d9ddccb2167bad34fc1f713ccc0866ca4c725c399ccb927011e584cd094266ee> (90)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/CompareImages/aws-lambda-tools-defaults.json<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-d64b9e15f91e11b955660d197d107feb25a6e0a9369837509b276993d85b4c67> (12)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/CompareImages/events/event.json<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-190dd448c9b72026d8a22d0b3882471ff694f63aeb2995250f28b061e39bd08d> (7)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/DomainModel/DomainModel.csproj<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-edc83802390fd02da90a61be5e4e2da3618a5058087a42df90065a2b47f92061> (17)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/DomainModel/EventBridgeEvent/Bucket.cs<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-e73d3137262d8c44bca37d356425e6a83c8887ff601350299c4f063fe144071f> (17)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/DomainModel/EventBridgeEvent/FaceComparisonDetails.cs<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-2ce8e3a2e3eb2df875cc7a88fe90552fec82834af76843a5dada87cadc9160d3> (33)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/DomainModel/EventBridgeEvent/S3Object.cs<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-ec81ff6f6e4755201cea9e97416ca7b516039005e2821ed13671eaaf52cf8e6b> (40)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/DomainModel/EventBridgeEvent/S3ObjectCreate.cs<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-d2f37787004ecf0954781335427d54d7c69e683bcef1ac76db29ef9c1ccbf976> (25)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/DomainModel/EventBridgeEvent/S3ObjectCreateEvent.cs<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-7ad460966bc289e48fedb4cf7edd9b76bdc9486aa85b32da453ec9c77fdd9568> (15)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/DomainModel/FaceComparison.cs<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-324c86913ba97158839e61daaa73d471425c40a13e7a6e562818d60c5946f2dd> (24)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/DomainModel/FaceComparisonRequest.cs<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-c24368cdfc0f1ec6c48d35fd4888d3912d84bd8a26fff51a4683b7fd9a1045c6> (28)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/GetTargetImages/Function.cs<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-ac08b671131538e321e3ff07b88738ad81f669d1dea5992153e8d2b2751a7179> (90)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/GetTargetImages/GetTargetImages.csproj<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-156b7004c7107345075b5490de978be312f1352183bb47a4469ad34f97734133> (28)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/GetTargetImages/aws-lambda-tools-defaults.json<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-119b1b9f5f5c24f49ee6ae9ff7e78477c7d7c63e56530092028c802bbf0da75f> (12)
  *   A Serverless/Serverless App with Dynamic Step Functions /functions/GetTargetImages/events/event.json<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-6f448d20ebf79af2c301bae160c86a180f7806d02d2049f1d7ec61097b842f8e> (29)
  *   A Serverless/Serverless App with Dynamic Step Functions /omnisharp.json<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-65fafde26cee4ad5e5d89a33dc9f3fa93ea18ad26ba857aaca5c0459a35bac87> (11)
  *   A Serverless/Serverless App with Dynamic Step Functions /samconfig.toml<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-b09e87ea96e227e034d45c0791ac2d8c1d1ae7d39e9e97a0f10555ed5e32e831> (11)
  *   A Serverless/Serverless App with Dynamic Step Functions /statemachine/compareFacesStateMachine.asl.json<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-a0fe5dd588fc1c784db370dfe4a3838b8d0f77472c9b6904fd89b69c592e83dc> (120)
  *   A Serverless/Serverless App with Dynamic Step Functions /stepfunctions_graph.svg<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-aef3b1071c968cd3b1ba3de5e0c95d671dd776c627abad3f67b77f5c7b9d0cdb> (4)
  *   A Serverless/Serverless App with Dynamic Step Functions /template.yaml<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-a508363dbcca53ab269fa69e657a080f3984c661dc3456ea2170adc6ebb12371> (120)
  *   A Serverless/Serverless App with Dynamic Step Functions /tests/CompareImages.Test/CompareImages.Test.csproj<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-0b6b076d309b19dede263d0a9acb5100b5fb87381c00707077a90f2c68d3746c> (20)
  *   A Serverless/Serverless App with Dynamic Step Functions /tests/CompareImages.Test/FunctionTest.cs<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-ff18c3d128729a551a83b1046f3b03ff1f57a6e80dbef7e284703074bd3d0f5f> (73)
  *   A Serverless/Serverless App with Dynamic Step Functions /tests/GetTargetImages.Test/FunctionTest.cs<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-7ea7d02fc8b6df3b1fa536fe12bcde73da6f17d53903fdd2c6c02327a36172fb> (106)
  *   A Serverless/Serverless App with Dynamic Step Functions /tests/GetTargetImages.Test/GetTargetImages.Test.csproj<https://github.com/aws-samples/aws-net-guides/pull/47/files#diff-03f64970aa61d1251873227d6258dfb5f5d68bb3581e8deed5cd5a1ad1101548> (19)

Patch Links:

  *   https://github.com/aws-samples/aws-net-guides/pull/47.patch
  *   https://github.com/aws-samples/aws-net-guides/pull/47.diff

—
Reply to this email directly, view it on GitHub<https://github.com/aws-samples/aws-net-guides/pull/47>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AEWAH3OVRHX64XUNSTTRWYTVXD5PLANCNFSM55KWQZBA>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.******@***.***>>

rezvanf commented:
Thank you for the feedback and review.  I have added the license and moved from absolute to relative paths.

In regards to the S3Objects this will initiate to an empty list in the event there is no objects in the s3 bucket. We do not need to check null here.
